cognite/client/data_classes/capabilities.py

```diff
@@ -176,31 +187,39 @@ def load(cls, resource: dict | str, allow_unknown: bool = False) -> Self:
             (name,) = known_acls
             capability_cls = _CAPABILITY_CLASS_BY_NAME[name]
             # TODO: We ignore extra keys that might be present like 'projectUrlNames' - are these needed?
+            try:
+                scopes = Capability.Scope.load(resource[name]["scope"], allow_unknown)
+            except Exception as err:
+                raise ValueError(f"Could not instantiate {capability_cls.__name__} due to: {err}") from None
+
             return cast(
                 Self,
                 capability_cls(
                     actions=[capability_cls.Action.load(act, allow_unknown) for act in resource[name]["actions"]],
-                    scope=Capability.Scope.load(resource[name]["scope"]),
+                    scope=scopes,
                     allow_unknown=allow_unknown,
                 ),
             )
-        elif not known_acls and len(resource) == 1:
-            # We infer this as an unknown acl not yet added to the SDK:
-            ((name, data),) = resource.items()
-            return cast(
-                Self,
-                UnknownAcl(
+        # We infer this as an unknown acl not yet added to the SDK. All API-responses are neccessarily
+        # loaded with 'allow_unknown=True' (to future-proof):
+        if allow_unknown:
+            if len(resource) != 1:
+                unknown_acl = UnknownAcl(actions=None, scope=None, raw_data=resource, allow_unknown=True)  # type: ignore [arg-type]
+            else:
+                ((name, data),) = resource.items()
+                unknown_acl = UnknownAcl(
                     capability_name=name,
-                    actions=[UnknownAcl.Action.load(act, allow_unknown) for act in data["actions"]],
-                    scope=Capability.Scope.load(data["scope"]),
+                    actions=[UnknownAcl.Action.load(act, allow_unknown=True) for act in data["actions"]],
+                    scope=Capability.Scope.load(data["scope"], allow_unknown=True),
                     raw_data=resource,
-                    allow_unknown=allow_unknown,
-                ),
-            )
-        # We got more than one acl (highly unlikely) or we got an unknown acl + extra keys in the response:
+                    allow_unknown=True,
+                )
+            return cast(Self, unknown_acl)
+
         raise ValueError(
-            "Unable to parse capability from API-response, please create an issue on Github: "
-            "https://github.com/cognitedata/cognite-sdk-python"
+            f"Unable to parse Capability, none of the top-level keys in the input, {sorted(resource)}, "
+            f"matched known ACLs, - or - multiple was found. Pass `allow_unknown=True` to force loading it "
+            f"as an unknown capability. List of known ACLs: {sorted(_CAPABILITY_CLASS_BY_NAME)}."
```

Comment:
Could we use the `builin` `difflib` to give a shorter l list the most relevant?

Reply:
The reason for this is that I think a typical issue here are typos, thus it would be useful to have a curated list of what could be the intintion.

Reply:
Something like the top 3 most similar ACLs by name?

Reply:
Makes sense to me, could also introduce and reference a constant that lists all?

Reply:
That is a great idea, then you avoid spam and allows the user to check it.